pyccl/tests/test_cclobject.py

```diff
@@ -0,0 +1,189 @@
+import pytest
+import numpy as np
+import pyccl as ccl
+import functools
+
+
+def test_CCLObject():
+    # Test eq --> repr <-- hash for all kinds of CCL objects.
+
+    # 1.1. Using a complicated Cosmology object.
+    extras = {"camb": {"halofit_version": "mead2020", "HMCode_logT_AGN": 7.8}}
+    kwargs = {"transfer_function": "bbks",
+              "matter_power_spectrum": "emu",
+              "z_mg": np.ones(10),
+              "df_mg": np.ones(10),
+              "extra_parameters": extras}
+    COSMO1 = ccl.CosmologyVanillaLCDM(**kwargs)
+    COSMO2 = ccl.CosmologyVanillaLCDM(**kwargs)
+    assert COSMO1 == COSMO2
+    kwargs["df_mg"] *= 2
+    COSMO2 = ccl.CosmologyVanillaLCDM(**kwargs)
+    assert COSMO1 != COSMO2
+
+    # 2. Using a Pk2D object.
+    cosmo = ccl.CosmologyVanillaLCDM(transfer_function="bbks")
+    cosmo.compute_linear_power()
+    PK1 = cosmo.get_linear_power()
+    PK2 = ccl.Pk2D.pk_from_model(cosmo, "bbks")
+    assert PK1 == PK2
+    assert ccl.Pk2D(empty=True) == ccl.Pk2D(empty=True)
```

Comment:
can you add a test when they're not the same?

Reply:
done